src/async_jobs.ts

```diff
@@ -42,6 +55,52 @@ async function slackInteractivityHandler(
     );
   }
 
+  if (
+    payload.type === 'shortcut' &&
+    payload.callback_id === SlackCallbackId.OPEN_CLOSE_CHANNELS
+  ) {
+    logger.info(
+      `SERVER POST /slack-interactivity: Determined user interaction is a OPEN_CLOSE_CHANNELS  shortcut.`
+    );
+
+    await SlackInteractionHandler.handleOpenCloseChannels({
+      payload,
+      redisClient,
+      originatingSlackUserName,
+      viewId: getViewId(payload, interactivityMetadata),
+    });
+    return;
+  }
+
+  // Handle open/close modal actions
+  if (
+    payload.type === 'block_actions' &&
+    payload.actions &&
+    [
+      SlackActionId.OPEN_CLOSE_CHANNELS_FILTER_STATE,
+      SlackActionId.OPEN_CLOSE_CHANNELS_FILTER_TYPE,
+      SlackActionId.OPEN_CLOSE_CHANNELS_CHANNEL_STATE_DROPDOWN,
+    ].includes(payload.actions[0].action_id as SlackActionId)
+  ) {
+    logger.info(
+      `SERVER POST /slack-interactivity: Determined user interaction is in OPEN_CLOSE_CHANNELS modal`
+    );
+
+    const view = payload.view;
+    if (!view) {
+      throw new Error('OPEN_CLOSE_CHANNELS block_actions expected view');
+    }
+
+    await SlackInteractionHandler.handleOpenCloseChannels({
+      payload,
+      redisClient,
+      originatingSlackUserName,
+      viewId: getViewId(payload, interactivityMetadata),
+      action: payload.actions[0],
+    });
+    return;
+  }
+
   if (payload.type === 'block_actions' || payload.type === 'message_action') {
```

Comment:
Something that might be nice for later is to simplify this function so it involves less nesting based on combinations of types and payloads and whatever. In theory, we should be able to do a giant switch statement on `action_id` and/or `callback_id`.